### PR TITLE
Make UV_HTTP_TIMEOUT default value bigger in WSL2 environment

### PR DESCRIPTION
**Pull request by potiuk**
When uv docker build is run in WSL2 the default timeout does not cope well with downloading big packages and times-out with the default UV timeout. This PR performs detection of whether you are in a WSL environment and changes the timeout in case you are in WSL2. It also raises error when WSL1 is detected (not supported by Breeze at all). In case the WSL1 detection is wrong we also have an escape hatch by manually specifying `--uv-http-timeout`.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
cc: @dabla

File: dev/breeze/src/airflow_breeze/utils/platforms.py [hunk 1]
```diff
@@ -16,10 +16,69 @@
...
+        "`--no-use-uv` flag to skip the WSL1 check.\n"
+    )
+
+
+def is_wsl2() -> bool:
+    """
+    Check if the current platform is WSL
+    :return: True if the current platform is WSL
+    """
+    release_name = platform.uname().release
```
Comment (Taragolis) [inline dev/breeze/src/airflow_breeze/utils/platforms.py:65]: If it not a linux we could stop check in very beginning

```suggestion
    if not sys.platform.startswith("linux"):
        return False
    release_name = platform.uname().release
```

Comment (potiuk) [reply dev/breeze/src/airflow_breeze/utils/platforms.py:65]: Why not :)

Comment (Taragolis) [reply dev/breeze/src/airflow_breeze/utils/platforms.py:65]: Yeah I guess it hardly possible that macOS pass over this condition `if not has_wsl_interop and not microsoft_in_release and not wsl_conf:`

But who knows what the changes happen in the future conditions

**Review by Taragolis — APPROVED** [review]

**Pull request merged**